geocity/apps/accounts/models.py

```diff
@@ -298,6 +304,41 @@ def create_anonymous_user(self):
                 zipcode=settings.ANONYMOUS_USER_ZIPCODE,
             )
 
+    def clean(self):
+        from geocity.apps.forms.models import Form
+        from geocity.apps.submissions.models import Submission
+
+        if (
+            Form.objects.annotate(entities_count=Count("administrative_entities"))
+            .filter(
+                is_public=True, entities_count__gt=1, administrative_entities=self.pk
+            )
+            .exists()
+        ):
+            raise ValidationError(
+                _(
+                    "Des formulaires partagés avec d'autres entités "
+                    "administratives sont encore disponibles."
+                )
+            )
+
+        if (
+            self.is_single_form_submissions
+            and Submission.objects.annotate(forms_count=Count("forms"))
+            .filter(forms_count__gt=1, administrative_entity=self.pk)
+            .exclude(status=Submission.STATUS_ARCHIVED)
+            .exists()
+        ):
+            raise ValidationError(
+                {
+                    "is_single_form_submissions": _(
```

Comment:
Why not put the message error at the same place as the other one?

Reply:
Because this one only appear when checking the boolean field "is_single_form_submissions".

To me, it makes sense, this one is a field error, the other one is a global error.

Reply:
+1 thanks